ocp_resources/virtual_machine_instance.py

```diff
@@ -204,11 +205,48 @@ def get_xml(self):
         Returns:
             xml_output(string): VMI XML in the multi-line string
         """
+        cmd = f"virsh {self.virt_launcher_pod_hypervisor_connection_uri} dumpxml {self.namespace}_{self.name}"
         return self.virt_launcher_pod.execute(
-            command=["virsh", "dumpxml", f"{self.namespace}_{self.name}"],
+            command=shlex.split(cmd),
             container="compute",
         )
 
+    @property
+    def virt_launcher_pod_user(self):
+        """
+        Get Virt Launcher Pod User value
+
+        Returns:
+            Int: Virt Launcher Pod runAsUser value
+        """
+        return self.virt_launcher_pod.instance.spec.securityContext.runAsUser
+
+    def is_virt_launcher_pod_root(self):
+        """
+        Check if Virt Launcher Pod is Root
+
+        Returns:
+            Bool: True if Virt Launcher Pod is Root.
+        """
+        return not bool(self.virt_launcher_pod_user)
+
+    @property
+    def virt_launcher_pod_hypervisor_connection_uri(self):
+        """
+        Get Virt Launcher Pod Hypervisor Connection URI
+
+        Required to connect to Hypervisor for
+        Non-Root Virt-Launcher Pod.
+
+        Returns:
+            String: Hypervisor Connection URI
+        """
+        return (
+            ""
+            if self.is_virt_launcher_pod_root
```

Comment:
Sonarcloud ftw. This needs to be a function call.

Reply:
Updated